lib/battle/CUnitState.cpp

```diff
@@ -407,6 +407,11 @@ int32_t CUnitState::getCasterUnitId() const
 	return static_cast<int32_t>(unitId());
 }
 
+BattleHex CUnitState::getCasterPosition() const
+{
+	return getPosition();
+}
+
```

Comment:
This won't works as intended for double-wide units. For example hex immediately behind unit would have distance of 1, even though it is adjacent to unit

Perhaps instead use interface like this?
```cpp
bool canCastAtTarget(const spells::Spell * spell, const BattleHex & hex)
```

For unit this would check distance, adjusted for double-wide units, and for heroes this would be empty method that always returns true

Reply:
Oh, even more - what if *target* is double-wide?

So this method should either accept an overload with battle::Unit as 2nd argument, or accept spell::Target (or what's its name) class